guess_the_words/function.py

```diff
@@ -0,0 +1,161 @@
+    choose_eser = input(f'Hello, {player_name}. Choose what you need:'
+                        f'\n1. Read rules of game - type "rules"'
+                        f'\n2. To start game - type "start"'
+                        f'\n3. To stop game - type "stop"'
+                        f'\n4. Show past result - type "result"')
+    return choose_eser
+
+
+def search_result(name_player: str) -> str:
+    with open('guess_result.json', 'r') as file:
+        data = json.load(file)
+        list_data = data['Result of games']
+        ii = 0
+        while ii < len(list_data):
+            list_j = []
+            ii += 1
+            for i, j in enumerate(list_data):
+                if name_player == j['Player name']:
+                    list_j.append(j)
+            return list_j
+
+
+def choose_loop(choose_user: str, player_name: str) -> str:
+    while True:
+        if choose_user == 'rules'.lower():
+            rules = rules_for_user()
+            print(rules)
+            break
+        elif choose_user == 'stop'.lower():
+            print('Game was stopped')
+            break
+        elif choose_user == 'start'.lower():
+            body_loop(player_name)
+            break
+        elif choose_user == 'result'.lower():
+            name_player = input('Enter player name: ').title()
+            result_search = search_result(name_player)
+            if result_search == []:
+                print('Player not defined. Try another name!')
+                break
+            else:
+                print(result_search)
+                break
+
+        else:
+            print('Incorrect input.')
+            break
+
+
+def val_alpha(player_letters: list) -> list:
+    if set(player_letters) - set(string.ascii_lowercase):
+        print("Unsupported symbols, use letters instead")
+        return True
+
+
+def result_add(count_dict):
+    with open('guess_result.json', 'r') as file:
+        data = json.load(file)
+        list_data = data['Result of games']
+        add_list = count_dict
+        list_data.append(add_list)
+    with open('guess_result.json', 'w') as file:
+        json.dump(data, file, indent=4)
+
+
+def body_loop(player_name):
+    count_dict = {'Date time game': '', 'Player name': player_name, 'Game result': 'Lose',
+                  'Score': {'Count tries': 0, 'Score points': 0}}
+
+    target_word = random.choice(create_list())
+
+    i = 0
+    while i < 6:
+        player_letters = list(input(f'Attempt {i + 1} of 6: ').lower())
+
+        if val_alpha(player_letters):
+            continue
+
+        if val_count(player_letters):
+            continue
+
+        target_letters = list(target_word)
+
+        for position, letter in enumerate(player_letters):
+            if letter == target_letters[position]:
+                player_letters[position] = '!'
+                target_letters[position] = None  # Mark as "found"
+
+        target_letters = [i for i in target_letters if i]
```

Comment:
Suggestion for my program :)
```suggestion
        target_letters = [i for i in target_letters if i is not None]
```

Reply:
I lost this 'i is not None' in my draft this part was. But I did use it in work function - deleted